src/guide/deployment/mongodb.md

```diff
@@ -0,0 +1,155 @@
+# How to deploy a MongoDB
+
+We provide packages for the following versions of MongoDB:
+
+| Version | Store Path | Platform release |
+| --- | --- | --- |
+| 4.0.27 | `/nix/store/jg8mgbfkyg9vbc68dwizy7vj92gzzsvs-mongodb-4.0.27` | 2024_001/22.11 |
+| 4.2.24 | `/nix/store/86f2ff9w16whjmq3y0c6aywmxxd27mva-mongodb-4.2.24` | 2024_002/23.05 |
+
+NixOS has support for providing a MongoDB systemd service.
+It can be configured as follows in `/etc/local/nixos/mongodb.nix`:
+
+```nix
+{ config, lib, pkgs, ... }:
+let
+  inherit (config) fclib;
+
+  mcfg = config.services.mongodb;
+
+  listenAddresses =
+    fclib.network.lo.dualstack.addresses ++
+    fclib.network.srv.dualstack.addresses;
+
+  checkPkg = pkgs.fc.check-mongodb;
+  checkMongoCmd = "${checkPkg}/bin/check_mongodb";
+  extraCheckArgs = "-h localhost -p 27017";
+
+  mongoCnf = pkgs.writeText "mongodb.yaml" ''
+    net.bindIp: "${mcfg.bind_ip}"
+    net.ipv6: true
+
+    ${lib.optionalString mcfg.quiet "systemLog.quiet: true"}
+    systemLog.path: /var/log/mongodb/mongodb.log
+    systemLog.destination: file
+    systemLog.logAppend: true
+    systemLog.logRotate: reopen
+
+    storage.dbPath: ${mcfg.dbpath}
+
+    processManagement.fork: true
+    processManagement.pidFilePath: ${mcfg.pidFile}
+
+    ${lib.optionalString (mcfg.replSetName != "") "replication.replSetName: ${mcfg.replSetName}"}
+
+    ${mcfg.extraConfig}
+  '';
+in {
+  environment.systemPackages = [
+    pkgs.mongodb-tools
+  ];
+
+  services.mongodb = {
+    enable = true;
+    dbpath = "/srv/mongodb";
+    bind_ip = fclib.mkPlatform (lib.concatStringsSep "," listenAddresses);
+    pidFile = "/run/mongodb.pid";
+    package = throw "Please replace this `throw ...` expression with 'builtins.storePath /nix/store/…'";
+    enableAuth = true;
+
+    # WARNING: this will only work on the very first install.
+    # You need to change MongoDB's root password manually later on.
+    initialRootPassword = throw "Please replace this `throw ...` expression with the initial root password of the mongodb root account";
+
+    # WARNING: don't set this on the initial install of MongoDB!
+    # Otherwise the initialRootPassword won't be set.
+    #replSetName = "replica";
+  };
+
+  systemd.services.mongodb = {
+    preStart = "echo never > /sys/kernel/mm/transparent_hugepage/defrag";
+    postStop = "echo always > /sys/kernel/mm/transparent_hugepage/defrag";
+    # intial creating of journal takes ages
+    serviceConfig.TimeoutStartSec = fclib.mkPlatform 1200;
```

Comment:
I would not use mkPlatform here … this is user documentation not platform code …